src/attack.c

```diff
@@ -1074,8 +1074,18 @@ attack_network (struct scan_globals *globals)
   max_checks = get_max_checks_number ();
 
   hosts = gvm_hosts_new (hostlist);
-  unresolved = gvm_hosts_resolve (hosts);
+  if (hosts == NULL)
+    {
+      sprintf (buf, "Invalid target list: %s.", hostlist);
```

Comment:
Buffer is very small (only 96 bytes) and `hostlist` can get very big. Use `snprinf` or `g_snprintf()` and a bigger buffer if its likely that hostlist will exceed 96 bytes length. You can use `strlen` to get the needed lenght and check it agains the buffer size.
Do we even limit the lenght of a hoststring?